src/contrib/transports/Akka.Remote.Transport.Streams.Tcp/Class1.cs

```diff
@@ -0,0 +1,313 @@
+                        .Recover(
+                            ex =>
+                            {
+                                handle.Notify(
+                                    new Disassociated(DisassociateInfo.Unknown));
+                                return IO.ByteString.Empty;
+                            }), _mat);
+                    queuePromise.SetResult(outQueue);
+                    listener.Notify(new InboundAssociation(handle));
+                });
+                return NotUsed.Instance;
+            }).ToMaterialized(Sink.Ignore<NotUsed>(),Keep.Left).Run(_mat);
+            return (_addr, AssociationListenerPromise);
+        }
+
+        
+        protected async Task<IPEndPoint> DnsToIPEndpoint(DnsEndPoint dns)
+        {
+            IPEndPoint endpoint;
+
+            var addressFamily = Settings.DnsUseIpv6 ? AddressFamily.InterNetworkV6 : AddressFamily.InterNetwork;
+            endpoint = await DnsHelpers.ResolveNameAsync(dns, addressFamily).ConfigureAwait(false);
+
+            return endpoint;
+        }
+        public override bool IsResponsibleFor(Address remote)
+        {
+            return true;
+        }
+
+        public override async Task<AssociationHandle> Associate(Address remoteAddress)
+        {
+            
+            var c =base.System.TcpStream().OutgoingConnection(
+                DnsHelpers.AddressToSocketAddress(remoteAddress),
+                _outAddr,
+                ImmutableList.Create<Inet.SocketOption>(
+                    new Inet.SO.ReceiveBufferSize(
+                        Settings.ReceiveBufferSize ?? 256 * 1024),
+                    new Inet.SO.SendBufferSize(
+                        Settings.SendBufferSize ?? 256 * 1024)));
+            
+            
+            var queuePromise =
+                    new TaskCompletionSource<
+                        ISourceQueueWithComplete<IO.ByteString>>();
+            var handle = new StreamingTcpAssociationHandle(_addr,remoteAddress, queuePromise);
+            handle.ReadHandlerSource.Task.ContinueWith(s =>
+            {   
+                var listener = s.Result;
+                handle.RegisterListener(listener);
+            }, TaskContinuationOptions.ExecuteSynchronously);
+            var run = c.JoinMaterialized(buildFlow(handle).Recover(
+                    ex =>
+                    {
+                        handle.Notify(
+                            new Disassociated(DisassociateInfo.Unknown));
+                        return IO.ByteString.Empty;
+                    }),(oc,sq)=>(oc,sq)).Run(_mat);
+            queuePromise.SetResult(run.sq);
+            return handle;
+        }
+
+        
+        private Source<IO.ByteString, ISourceQueueWithComplete<IO.ByteString>> sendFlow()
+        {
+            //TODO: Improve batching voodoo in pipeline.
+            return Source
+                .Queue<IO.ByteString>(512, OverflowStrategy.DropNew)
```

Comment:
Probably better for us if we just reject the new write to the queue, so the `EndpointWriter` can backoff

Reply:
@Aaronontheweb my bad on this, you want to look at /src/core/akka.remote/  for the most up to date bits.

Moving everything into its own /contrib/ will be a bit more work because of all the `internal` things a transport needs

Reply:
I think this is a great start - I’m very impressed with your hard work. I think this will be an amazing improvement over DotNetty

Reply:
Looking forward to reading the rest of it